Netlify Functions & Fetch Instagram API (#364)
Coding-Coach/coding-coach: content/blog/netlify-serverless-function-instagram-api/index.md

---
title: Instagram Feed API
date: 2019-11-27
tags:
  - instagram
  - lambda
  - netlify
image: netlify-lambda.jpg
author: cristian-echeverria
---

While I was working on a website for a relative, one of the requirements was to link one of the sections of the site with their Instagram account.

So the concept was that; every time they post a picture on Instagram shows that picture on their website. But, only show the last 3 pictures.

> So I was Googlelin trying to find a solution for this, I was thinking there has to be a third-party solution for this, and maybe I can find it for FREE, and yes, I found a couple of solutions but none was something I liked.

Reviewer: ```suggestion
So I was googling trying to find a solution for this, I was thinking there has to be a third-party solution for this, and maybe I can find it for FREE, and yes, I found a couple of solutions but none was something I liked.
```

Now that we have access to AWS Lambda Functions we can use one of the examples that you can find in Netlify site and create a Javascript script in your project like this:

```
const axios = require('axios')

exports.handler = function instagram(event, context, callback) {
  const endpoint = 'https://api.instagram.com/v1/users/self/media/recent'
  const token = process.env.INSTAGRAM_ACCESS_TOKEN
  const limit = 5

  axios
    .get(`${endpoint}?access_token=${token}&count=${limit}`)
    .then(({ data: { data: posts } }) => {
      callback(null, {
        statusCode: 200,
        headers: {
          'content-type': 'application/json',
        },
        body: JSON.stringify(
          posts.map(i => ({
            id: i.id,
            link: i.link,
            images: i.images,
            videos: i.videos,
            caption: i.caption.text,
          })),
        ),
      })
    })
    .catch((e) => {
      callback(e)
    })
}
```

### Step 3: Testing your Lambda

As you can see, at this point we need to install some node libraries like Axios in order to fetch data from Instagram API. Also, you can test your code locally using netlify-lambda.

`npm i netlify-lambda axios`

> With that, you can add a command into your package.json in order to serve your lambda function using netlify-lambda, you have to specify the location of your lambda function. In my case I created a `./functions` folder and added my lambda functions there:

Reviewer: ```suggestion
With that, you can add a new script into your `package.json`. In order to serve your lambda function using `netlify-lambda`, you have to specify the location of your lambda function. In my case I created a `./functions` folder and added my lambda functions there:
```

```
"scripts": {
    "lambda-serve": "netlify-lambda serve functions",
    "lambda-build": "netlify-lambda build functions"
  },
```

> Another useful step is create a `netlify.toml` file in the root of your project and you can add the some commands like this:

Reviewer: ```suggestion
Another useful step is create a `netlify.toml` file in the root of your project, then you can add the following commands:
```

```
[build]
    functions = "lambda"
    Command = "npm run lambda-build"
```

> Have in mind that you can also [specify the location of your lambda functions](https://docs.netlify.com/functions/configure-and-deploy/#configure-the-functions-folder) in netlify settings for your project, but if you use **netlify.toml** file it will override your configuration.

Reviewer: ```suggestion
Keep in mind that you can also [specify the location of your lambda functions](https://docs.netlify.com/functions/configure-and-deploy/#configure-the-functions-folder) in netlify settings for your project, but if you use **netlify.toml** file it will override your configuration.
```
